### ISSUE TITLE
Translation for PageSettings : Themes and Theme Containers is not working.

### ISSUE THREAD

**Issue by leigh-pointer** [oqtane/oqtane.framework#1932]
The Localizer is looking in the incorrect folder.  The path it should be looking at is:
Oqtane.Resources.Themes.OqtaneTheme.Themes.ThemeSettings

Localizer
{Microsoft.Extensions.Localization.StringLocalizer<Oqtane.Themes.OqtaneTheme.ThemeSettings>}

Localizer["Login.Text"]
{Login.Text}
    Name: "Login.Text"
    ResourceNotFound: true
    SearchedLocation: "Oqtane.Resources.Themes.OqtaneTheme.ThemeSettings"
    Value: "Login.Text"

Also not woring for the "ResourceKey" localizations.

This for all languages including the internal resources.

**Comment by hishamco** [comment]
I may have a look to this

**Comment by leigh-pointer** [comment]
@hishamco 
> I may have a look to this
Cool.

The only thing that looks like if it could be an issue is that **Themes** appears twice in the Resx path and in the Localizer it only appear twice.  The path for the resx is correct.

**Comment by sbwalker** [comment]
I investigated this issue...

The location of RESX files is based on the namespace of a component - not the location of a razor file on disk. This is by design. The namespace of the ThemeSettings component is Oqtane.Themes.OqtaneTheme.ThemeSettings - therefore the RESX file should be located \Resources\Themes\OqtaneTheme\ThemeSettings.resx. However after I moved these RESX files, localization was still not working properly for the Label control.

The Label control inherits from LocalizableComponent - and internally it uses the cascading parameter ModuleState.ModuleType to determine the type name of the component being localized. This works fine for standard module components. However for components that are dynamically embedded inside of another module component - such as Module Settings, Theme Settings, Container Settings - it does not work as it is not able to determine the type name correctly. I have fixed this by adding an additional parameter named ResourceType which can be utilized in these scenarios to explicitly set the type name of the component.

I will submit the PR shortly which resolves this issue.
